### PR TITLE
Gravity slider

### PR DESCRIPTION
**Pull request by geralddejong**

### CODE REVIEW

File: src/controls/mod.rs [hunk 1]
```diff
@@ -269,6 +285,11 @@ impl Program for ControlState {
                     self.action_queue.borrow_mut().push(action);
                 }
             }
+            Message::Gravity(message) => {
+                if let Some(action) = self.gravity_control.update(message) {
+                    self.action_queue.borrow_mut().push(action);
```
Comment (mitchelljustin) [inline src/controls/mod.rs:290]: You should extract a method `queue_action()` for this

File: src/experiment.rs [hunk 2]
```diff
@@ -25,10 +27,14 @@ pub struct Experiment {
     add_pulls: Option<f32>,
 }
 
+impl Experiment {
+}
+
```
Comment (mitchelljustin) [inline src/experiment.rs:32]: ?

File: src/experiment.rs [hunk 3]
```diff
@@ -25,10 +27,14 @@ pub struct Experiment {
...
 impl Default for Experiment {
     fn default() -> Self {
         Self {
             fabric: Fabric::default(),
+            physics: AIR_GRAVITY.clone(),
```
Comment (mitchelljustin) [inline src/experiment.rs:37]: Do you have to clone here?

Comment (mitchelljustin) [reply src/experiment.rs:37]: Since it's a const and all

**Review by mitchelljustin — APPROVED** [review]
Few comments is all

File: src/controls/mod.rs (hunk 1 above)
Comment (geralddejong) [reply src/controls/mod.rs:290]: Indeed

File: src/experiment.rs (hunk 2 above)
Comment (geralddejong) [reply src/experiment.rs:32]: oopsie.  cruft left after letting the IDE create the method.

File: src/experiment.rs (hunk 3 above)
Comment (geralddejong) [reply src/experiment.rs:37]: Within the experiment, it has to be mutated.

**Pull request merged**